ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -582,6 +596,93 @@ func (c Client) CreateTx(goCtx context.Context, account cosmosaccount.Account, m
 	}, nil
 }
 
+// GetBlockTXs returns the transactions in a block.
+// The list of transactions can be empty if there are no transactions in the block
+// at the moment this method is called.
+// Tendermint might index a limited number of block so trying to fetch transactions
+// from a block that is not indexed would return an error.
+func (c Client) GetBlockTXs(ctx context.Context, height int64) (txs []TX, err error) {
+	if height == 0 {
+		return nil, ErrInvalidBlockHeight
+	}
+
+	r, err := c.RPC.Block(ctx, &height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch block %d: %w", height, err)
+	}
+
+	query := createTxSearchByHeightQuery(height)
+
+	// TODO: improve to fetch pages in parallel (requires fetching page 1 to calculate n. of pages)
+	page := 1
+	perPage := defaultTXsPerPage
+	blockTime := r.Block.Time
+	for {
+		res, err := c.RPC.TxSearch(ctx, query, false, &page, &perPage, orderAsc)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tx := range res.Txs {
+			txs = append(txs, TX{
+				BlockTime: blockTime,
+				Raw:       tx,
+			})
+		}
+
+		// Stop when the last page is fetched
+		if res.TotalCount <= (page * perPage) {
+			break
+		}
+
+		page++
+	}
+
+	return txs, nil
+}
+
+// CollectTXs collects transactions from multiple consecutive blocks.
+// Transactions from a single block are send to the channel only if all transactions
+// from that block are collected successfully.
+// Blocks are traversed sequentially starting from a height until the latest block height
+// available at the moment this method is called.
+// The channel might contain the transactions collected successfully up until that point
+// when an error is returned.
+func (c Client) CollectTXs(ctx context.Context, fromHeight int64, tc chan<- []TX) error {
+	defer close(tc)
+
+	status, err := c.Status(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch latest block height: %w", err)
+	}
```

Comment:
@jeronimoalbi if you only need the latest block height, you can use `c.LatestBlockHeight()`

Reply:
Oh, I didn't know. Nice, I will change it.

Reply:
Done in [e0e1773](https://github.com/ignite/cli/pull/2528/commits/e0e1773a253b91a79054388e3f2a74907522c4c1)